src/spikeinterface/core/job_tools.py

```diff
@@ -223,6 +223,9 @@ def ensure_chunk_size(recording, total_memory=None, chunk_size=None, chunk_memor
     else:
         if n_jobs == 1:
             # not chunk computing
+            #TODO Discuss, Sam, is this something that we want to do?
+            # Even in single process mode, we should chunk the data to avoid loading the whole thing into memory I feel
+            # Am I wrong?
             chunk_size = None
```

Comment:
@samuelgarcia I realize that when n_jobs=1 set chunk_size = None
Whicn inside the function `ensure_chunk_size` ends up causing that the whole traces are extracted per segment:

```
def divide_segment_into_chunks(num_frames, chunk_size):
    if chunk_size is None:
        chunks = [(0, num_frames)]
```

Do we want this behavior? My feelign is that it shouldn't as indicated in the comment ot the right
but ...
If so, I really think this should not happen in such an obscure way (hidden deep in two functions). 
Let me know what do you think.

Reply:
The on purpose chunk_size=None have the meaning of do it with no chunking.
Forcing this when n_job=1 is bad idea! I do not remember why we have this here I guess it come from the previous framework logic to keep some compatibility.

Reply:
Fortunatly this code zone should be reach any more with the recent get_global_job_kwargs() mechanism.

Reply:
OK. I am removing this.

I am not sure this is working everywhere but we will see. For that to happen the function `fix_job_kwargs` should have been called which in turn  fetches the job_kwargs from globals and updates if neede. This is done in most places but I am not sure I did it when writing code that takes job_kwargs....

I find this behavior brittle. I think it would be way easier to have that as default values of the ChunkRecordingExecutor...

Reply:
Suggestion:
This last case could be covered by:
- `chunk_size=a big number`
- `chunk_duration="full"`

And we should check that at least one of `chunk_size`, `chunk_duration`, `chunk_memory` or `total_memory` is not None.